### ISSUE TITLE
can not highlight after a list comprehension 

### ISSUE THREAD

**Issue by Babystepstocoding** [JustusAdam/language-haskell#177]
<img width="348" alt="Screen Shot 2020-11-10 at 12 28 13 pm" src="https://user-images.githubusercontent.com/57067027/98616113-36485d00-2350-11eb-89f5-bc611c4c56ce.png">

**Comment by sheaf** [comment]
Hi, thanks for the bug report.    

This problem is due to a conflict between list comprehension and quasi-quotations. To fix the highlighting, you need to add a space around x, e.g.

```haskell
[ x | x<-list, p x ]
```

Note that the same issue exists on GitHub:

```haskell
-- using list comprehension
myFilter :: (a-> Bool) -> [a] -> [a]
myFilter p list  [x| x<- list, px]

-- not highlighted
f :: x -> x
f x = x
--
{-

-}

```

**Comment by sheaf** [comment]
Closed as duplicate of #176.

**Comment by Babystepstocoding** [comment]
Thanks for your email. I posted this one on my school Piazza. Thanks.
